src/OrbitGl/TrackContainer.cpp

```diff
@@ -370,11 +369,66 @@ void TrackContainer::SetIteratorOverlayData(
   RequestUpdate(RequestUpdateScope::kDraw);
 }
 
+void TrackContainer::DrawThreadDependency(PrimitiveAssembler& primitive_assembler,
+                                          PickingMode picking_mode) {
+  std::optional<orbit_client_data::ThreadStateSliceInfo> thread_state_slice =
+      app_->selected_thread_state_slice();
+  if (!thread_state_slice.has_value() || picking_mode != PickingMode::kNone) {
+    return;
+  }
+
+  orbit_client_data::ThreadID start_arrow_thread = thread_state_slice->wakeup_tid();
+  orbit_client_data::ThreadID end_arrow_thread = thread_state_slice->tid();
+
+  std::optional<ThreadTrack*> temp_start_arrow_track =
+      track_manager_->GetThreadTrack(start_arrow_thread);
+  std::optional<ThreadTrack*> temp_end_arrow_track =
+      track_manager_->GetThreadTrack(end_arrow_thread);
+
+  if (!temp_end_arrow_track.has_value() || !temp_start_arrow_track.has_value()) {
+    return;
+  }
+
+  ThreadTrack* start_arrow_track = temp_start_arrow_track.value();
+  ThreadTrack* end_arrow_track = temp_end_arrow_track.value();
+
+  const std::vector<Track*>& threads = track_manager_->GetVisibleTracks();
```

Comment:
As it's not at all a responsibility of TrackManager, track_manager_->VisibleTracks() doesn't have any relationship with visibility in screen. 

It is a confused name and might need to be refactored, but it includes all the tracks that should be drawn (not filtered by any visibility option - we can filter by name or by type -). For example that list is including Tracks outside of the visible screen part.

Please:
- Double-check what happens if one of the tracks is out of the visible part of the screen (only the body of the arrow should be visible in my opinion).
- Try to filter a Track that holds the arrow to see if the behavior is consistent.
- Try to move a Track that holds an arrow.

Reply:
I don't use VisibleTracks to check whether a track is in the screen. I use it to check if a thread has a thread state bar. Basically I wanna see if `start_arrow_track` and `end_arrow_track` have thread state bars in the capture window.

I have tried all the three cases you mentioned before and they all work correctly.